provider/cmd/pulumi-gen-eks/main.go

```diff
@@ -300,12 +306,18 @@ func generateSchema() schema.PackageSpec {
 						Description: "Optional mappings from AWS IAM users to Kubernetes users and groups.",
 					},
 					"vpcCniOptions": {
-						TypeSpec: schema.TypeSpec{Ref: "#/types/eks:index:VpcCniOptions"},
+						TypeSpec: schema.TypeSpec{
+							Ref:   "#/types/eks:index:VpcCniOptions",
+							Plain: true,
+						},
```

Comment:
This is plain in the implementation:

https://github.com/pulumi/pulumi-eks/blob/c6f1fce98fa2683abe7ffbad7f166da99928fb83/nodejs/eks/cluster.ts#L981-L985